src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala

```diff
@@ -68,18 +70,14 @@ package object servlets extends DebugEnhancedLogging {
 
   implicit class RichZipInputStream(val zipInputStream: ZipInputStream) extends AnyVal {
     def unzipPlainEntriesTo(dir: File): Try[Unit] = {
-      def extract(entry: ZipEntry) = {
+      def extract(entry: ZipEntry): Try[Any] = {
         if (entry.isDirectory)
           Try((dir / entry.getName).createDirectories())
         else {
-          if (entry.getName.matches(extensionZipPattern))
-            Failure(BadRequestException(s"ZIP file is malformed. It contains a Zip ${ entry.getName }."))
-          else {
-            logger.info(s"Extracting ${ entry.getName } size=${ entry.getSize } compressedSize=${ entry.getCompressedSize } CRC=${ entry.getCrc }")
-            Try(Files.copy(zipInputStream, (dir / entry.getName).path))
-          }.recoverWith {
-            case e if e.isInstanceOf[ZipException] => Failure(BadRequestException(s"ZIP file is malformed. $e"))
-          }
+          logger.info(s"Extracting ${ entry.getName } size=${ entry.getSize } compressedSize=${ entry.getCompressedSize } CRC=${ entry.getCrc }")
+          Try(Files.copy(zipInputStream, (dir / entry.getName).path))
+        }.recoverWith {
```

Comment:
is this `recoverWith` applicable to the `else` branch only, or also to the `if` branch? That isn't clear to me from the code. Not sure if you have the right functionality now or if you're introducing a bug in error handling.

Reply:
I guess only to the else clause. But i assume 
`Try((dir / entry.getName).createDirectories())` is not able to cause a ZipException